xc/device.ml

```diff
@@ -538,6 +547,12 @@ module Vbd_Common = struct
       | Disk -> "disk"
       | Floppy -> "floppy";
     ];
+    (* Hack: this should be returned separately from SMAPIv3 attach call *)
+    let (params, qemu_params) = if Astring.String.is_prefix ~affix:"hack|" x.params then begin
+        match String.split_on_char '|' x.params with
```

Comment:
Just making sure: this does not filter empty strings. Is that the expected behaviour?

Reply:
Yes, this is the correct behaviour because of how SM constructs this string:
```
params = "hack|%s|%s" % (params, qemu_params) 
```